python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+
+            # This is just to make sure there are at least some images when we expect
+            # and the previous line is not just skipping everything.
+            images_tested += 1
+
+            lens = df.query('twinkles_sysno == %i' % spr_sys_df['twinkles_system'].iloc[0])
+            # SNe systems might not have all images appearing in an instance catalog unlike AGN
+            img_vals = spr_sys_df['image_number'].values
+            
+            for img_idx in range(len(img_vals)):
+                # Calculate the offsets from the lens galaxy position
+                offset_x1, offset_y1 = self.offset_on_sky(spr_sys_df['raPhoSim'].iloc[img_idx], 
+                                                          spr_sys_df['decPhoSim'].iloc[img_idx],
+                                                          lens_gal_df['raPhoSim'],
+                                                          lens_gal_df['decPhoSim'])
+
+                x_offsets.append(offset_x1-lens['x'].iloc[img_vals[img_idx]])
+                y_offsets.append(offset_y1-lens['y'].iloc[img_vals[img_idx]])
+                total_offsets.append((np.sqrt(x_offsets[-1]**2. + y_offsets[-1]**2.)/
+                                      np.sqrt(lens['x'].iloc[img_vals[img_idx]]**2. + 
+                                              lens['y'].iloc[img_vals[img_idx]]**2.)))
+
+        max_total_err = np.max(total_offsets)
+
+        print('------------')
+        print('SNe Location Test Results:')
+
+        print('Tested %i systems with images.' % images_tested)
+
+        if max_total_err < 0.01:
+            print('Pass: Max image offset error less than 1 percent')
+        else:
+            raise CatalogError('\nFail: Max image offset error greater than 1 percent. ' + 
+                               'Max total offset error is: %.4f percent.' % max_total_err)
+
+        print('------------')
+
+        return
+
+    def compare_agn_inputs(self, spr_agn_df, spr_agn_lens_df):
+
+        """
+        This test compares the things that get swapped in directly from the input catalog.
+        """
+        
+        db = om10.DB(catalog=os.path.join(os.environ['TWINKLES_DIR'], 'data', 
+                                          'twinkles_lenses_v2.fits'), vb=False)
+
+        errors_present = False
+        phi_error = False
+        z_lens_error = False
+        z_src_error = False
+        lens_major_axis_error = False
+        lens_minor_axis_error = False
+        lens_sed_error = False
+        errors_string = "Errors in: "
+
+        for lens_gal_row in spr_agn_lens_df.iterrows():
+
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_agn_df.query('lens_galaxy_uID == %i' % u_id)
+            use_lens = db.lenses['LENSID'][np.where(db.lenses['twinklesId'] == 
+                                                    spr_sys_df['twinkles_system'].iloc[0])[0]]
+            lens = db.get_lens(use_lens)
+
```

Comment:
Should all of these thresholds be in `np.abs()`? 

i.e. what if `lens['PHIE']+lens_gal_df['positionAngle'] = -10.0`?